Mathlib/RingTheory/AlgebraicIndependent/Basic.lean

```diff
@@ -130,6 +146,20 @@ theorem AlgHom.algebraicIndependent_iff (f : A →ₐ[R] A') (hf : Injective f)
 theorem algebraicIndependent_of_subsingleton [Subsingleton R] : AlgebraicIndependent R x :=
   algebraicIndependent_iff.2 fun _ _ => Subsingleton.elim _ _
 
+lemma isTranscendenceBasis_iff_of_subsingleton [Subsingleton R] (x : ι → A) :
+    IsTranscendenceBasis R x ↔ Nonempty ι := by
+  have := Module.subsingleton R A
+  refine ⟨fun h ↦ ?_, fun h ↦ ⟨algebraicIndependent_of_subsingleton,
+    fun s hs hx ↦ hx.antisymm fun a _ ↦ ⟨Classical.arbitrary _, Subsingleton.elim ..⟩⟩⟩
+  by_contra hι; rw [not_nonempty_iff] at hι
+  have := h.2 {0} algebraicIndependent_of_subsingleton
+  simp [range_eq_empty, eq_comm (a := ∅)] at this
+
+theorem trdeg_subsingleton [Subsingleton R] : trdeg R A = 1 :=
+  have := Module.subsingleton R A
```

Comment:
Maybe mark this as `@[nontriviality]`? Perhaps same as the above `isTranscendenceBasis_iff_of_subsingleton`.